### PR TITLE
ROX-23952: Add compliance report manager

### PR DESCRIPTION
**Pull request by ksurabhi91**
This PR adds compliance report manager to
-queue requests
-read from queue and submit job for report generation
-fixes bug with stop routine([ROX-24185](https://issues.redhat.com/browse/ROX-24185))

Testing Done
Added unit tests 
E2E tests in CI have passed

### CODE REVIEW

File: central/complianceoperator/v2/scanconfigurations/service/singleton.go [hunk 1]
```diff
@@ -23,5 +24,9 @@ func Singleton() Service {
 	serviceInstanceInit.Do(func() {
 		serviceInstance = New(scanSettingsDS.Singleton(), scanSettingBindingsDS.Singleton(), suiteDS.Singleton(), compliancemanager.Singleton())
 	})
+	if features.ComplianceReporting.Enabled() {
+		reportManager := reportManager.Singleton()
```
[the remark this answers is not in the document]
Comment (clickboo) [reply central/complianceoperator/v2/scanconfigurations/service/singleton.go:28]: Also I would add the start before service instance init

**Review by clickboo — APPROVED** [review]
🚢  after CI green. Thanks for the perseverance on this!

**Comment by ksurabhi91** [comment]
//test gke-qa-e2e-tests

**Review by dashrews78 — APPROVED** [review]
Obviously pending a green CI and verification the TLS test was indeed executed.

**Pull request merged**